wagyu/cli/ethereum.rs

```diff
@@ -58,13 +58,11 @@ impl EthereumWallet {
         })
     }
 
-    pub fn new_hd<N: EthereumNetwork, W: EthereumWordlist, R: Rng>(
-        rng: &mut R,
-        word_count: u8,
+    pub fn new_hd<N: EthereumNetwork, W: EthereumWordlist>(
+        mnemonic: &EthereumMnemonic<N, W>,
```

Comment:
This should be reverted. A `new_hd` wallet is supposed to be a randomly sampled HD wallet, __NOT__ a deterministic wallet. There is a method called `from_mnemonic` for this exact purpose.